test/jdk/java/util/zip/GZIP/GZIPInputStreamAvailable.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 7036144
+ * @summary Test concatenated gz streams when available() returns zero
+ */
+
+import java.io.*;
+import java.util.*;
+import java.util.zip.*;
+
+public class GZIPInputStreamAvailable {
+
+    public static void main(String [] args) throws IOException {
+
+        // Create gz data
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try (GZIPOutputStream out = new GZIPOutputStream(baos)) {
+            out.write("boo".getBytes("ASCII"));
+        }
+        final byte[] gz = baos.toByteArray();
+
+        // Repeat 32 times
+        baos.reset();
+        for(int i = 0; i < 32; i++)
+            baos.write(gz);
+        final byte[] gz32 = baos.toByteArray();
+
+        // (a) Read it from a stream where available() is accurate
+        final long count1;
+        try (GZIPInputStream in = new GZIPInputStream(new ByteArrayInputStream(gz32))) {
+            count1 = countBytes(in);
+        }
```

Comment:
Consider rewriting countBytes to take the ByteArrayInputStream/ZeroAvailableInputStream as a parameter, so you could just do:

 ```suggestion
        long count1 = countBytes(new ByteArrayInputStream(gz32));
```

Reply:
Thanks - should be addressed in c7087e55319.